doc/src/devdocs/build/build.md

```diff
@@ -239,10 +241,32 @@ For packaging Julia with LLVM, we recommend either:
  - bundling a Julia-only LLVM library inside the Julia package, or
  - adding the patches to the LLVM package of the distribution.
    * A complete list of patches is available in on [Github](https://github.com/JuliaLang/llvm-project) see the `julia-release/15.x` branch.
-   * The only Julia-specific patch is the lib renaming (`llvm-symver-jlprefix.patch`), which should _not_ be applied to a system LLVM.
+   * The only Julia-specific patch is the lib renaming (`llvm7-symver-jlprefix.patch`), which should _not_ be applied to a system LLVM.
    * The remaining patches are all upstream bug fixes, and have been contributed into upstream LLVM.
 
-Using an unpatched or different version of LLVM will result in errors and/or poor performance. Though Julia can be built with newer LLVM versions, support for this should be regarded as experimental and not suitable for packaging.
+Using an unpatched or different version of LLVM will result in errors and/or poor performance.
+You can build a different version of LLVM from a remote Git repository with the following options in the `Make.user` file:
+
+```make
+# Force source build of LLVM
+USE_BINARYBUILDER_LLVM = 0
+# Use Git for fetching LLVM source code
+DEPS_GIT = 1
+# URL of the Git repository you want to obtain LLVM from:
+LLVM_GIT_URL = ...
+# Name of the branch to check out automatically
+LLVM_BRANCH = standalone_rel_15_x
+LLVM_SHA1 = $(LLVM_BRANCH)
```

Comment:
I think with `DEPS_GIT` you don't need to set the `SHA1`

Reply:
If I don't set that  get
```
cd /home/mose/repo/julia/deps/srccache/llvm && git checkout -q --detach julia-15.0.7-5
fatal: git checkout: --detach does not take a path argument 'julia-15.0.7-5'
/home/mose/repo/julia/deps/llvm.mk:9: recipe for target '/home/mose/repo/julia/deps/srccache/llvm/source-extracted' failed
```

Reply:
It seems like at some point that the value for LLVM_SHA1 variable got changed to a branch name, instead of a commit hash, which is not as good for reproducibility. A branch is okay for development, but usually this should be a SHA1 (and then go checkout the branch you want manually).

Reply:
Technically the first variable there is the name of the reference to pull from the remote url, then the second is the reference to checkout.